api/tests/unit/projects/tags/test_views.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+from django.urls import reverse
+from pytest_lazyfixture import lazy_fixture
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from projects.models import Project
+from projects.tags.models import Tag
+
+
+@pytest.mark.parametrize(
+    "client",
+    [(lazy_fixture("admin_master_api_key_client")), (lazy_fixture("admin_client"))],
+)
```

Comment:
We're prioritizing using lower levels of access over the `admin_client` when it comes to checking permissions. So since this endpoint follows `TagPermissions` you'll want to use `staff_client` with the `with_project_permissions([VIEW_PROJECT])` to set access level to match. 

Also, it would be good to check the `HTTP 403` case to make sure the endpoint isn't open to a user from a different project.

Reply:
That's a good point… Although I wonder if that should be done by the unit test that test the permission class itself instead of the test that test the view?

Reply:
Since permissions can be misconfigured when adding to a viewset (which we've found wide open ones because of a decorator misordering) I think testing through the view overall is a better strategy.

Reply:
Yeah, that makes sense